From c770c4f52ede9042e9373cc2aa6d09b482a56028 Mon Sep 17 00:00:00 2001
From: Beata Michalska <beata.michalska@arm.com>
Date: Mon, 1 Sep 2025 13:30:52 +0200
Subject: [PATCH 052/161] rust: drm: file: Enable retrieveing client id.

Signed-off-by: Beata Michalska <beata.michalska@arm.com>
---
 rust/kernel/drm/file.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rust/kernel/drm/file.rs b/rust/kernel/drm/file.rs
index e8789c9110..401ca4bfec 100644
--- a/rust/kernel/drm/file.rs
+++ b/rust/kernel/drm/file.rs
@@ -53,6 +53,12 @@ pub fn inner(&self) -> Pin<&T> {
         unsafe { Pin::new_unchecked(&*(self.driver_priv())) }
     }
 
+    /// Return a unique id for fdinfo
+    pub fn get_client_id(&self) -> u64 {
+        // SAFETY: By the type invariants of `Self`, `self.as_raw()` is always valid.
+        unsafe { (*self.as_raw()).client_id }
+    }
+
     /// The open callback of a `struct drm_file`.
     pub(crate) extern "C" fn open_callback(
         raw_dev: *mut bindings::drm_device,
-- 
2.51.0

