From 6f053ecd7827833ddaf743399bdb32f6e2d58cf6 Mon Sep 17 00:00:00 2001
From: Daniel Almeida <daniel.almeida@collabora.com>
Date: Tue, 14 Oct 2025 19:57:56 -0300
Subject: [PATCH 116/161] rust: drm: syncobj: add find_fence()

---
 rust/kernel/drm/syncobj.rs | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/rust/kernel/drm/syncobj.rs b/rust/kernel/drm/syncobj.rs
index 734d200ae3..9265c76ffa 100644
--- a/rust/kernel/drm/syncobj.rs
+++ b/rust/kernel/drm/syncobj.rs
@@ -48,6 +48,37 @@ pub fn fence_get(&self) -> Option<Fence> {
         }
     }
 
+    /// Finds the fence at a specific timeline point for this sync object.
+    pub fn find_fence(
+        file: &drm::File<T::File>,
+        handle: u32,
+        point: u64,
+        flags: u64,
+    ) -> Result<Option<Fence>> {
+        let mut fence = core::ptr::null_mut();
+
+        // SAFETY: All arguments are valid per the type invariants
+        let ret = unsafe {
+            bindings::drm_syncobj_find_fence(
+                file.as_raw() as *mut _,
+                handle,
+                point,
+                flags,
+                &mut fence,
+            )
+        };
+
+        if ret != 0 {
+            Err(Error::from_errno(ret))
+        } else if fence.is_null() {
+            Ok(None)
+        } else {
+            // SAFETY: The pointer is non-NULL and drm_syncobj_find_fence acquired an
+            // additional reference.
+            Ok(Some(unsafe { Fence::from_raw(fence) }))
+        }
+    }
+
     /// Replaces the DMA fence with a new one, or removes it if fence is None.
     pub fn replace_fence(&self, fence: Option<&Fence>) {
         // SAFETY: All arguments should be valid per the respective type invariants.
-- 
2.51.0

